aggregator/prover/prover.go

```diff
@@ -2,81 +2,254 @@ package prover
 
 import (
 	"context"
+	"errors"
+	"fmt"
 	"time"
 
+	"github.com/0xPolygonHermez/zkevm-node/aggregator/pb"
+	"github.com/0xPolygonHermez/zkevm-node/config/types"
 	"github.com/0xPolygonHermez/zkevm-node/log"
-	proverclientpb "github.com/0xPolygonHermez/zkevm-node/proverclient/pb"
-	"google.golang.org/grpc"
-	"google.golang.org/grpc/connectivity"
-	"google.golang.org/grpc/credentials/insecure"
 )
 
-// Prover struct
+// Prover abstraction of the grpc prover client.
 type Prover struct {
-	URI    string
-	Client proverclientpb.ZKProverServiceClient
-	Conn   *grpc.ClientConn
+	id                                         string
+	IntervalFrequencyToGetProofGenerationState types.Duration
```

Comment:
Is this needed to be exported?

Reply:
not at all @ToniRamirezM :) , nice catch fbd90aa39452cf8dbe46c1a6549ad632cbdbf0d3